blog/assembly/contracts/main.ts

```diff
@@ -0,0 +1,53 @@
+import { Storage, callerHasWriteAccess, generateEvent } from '@massalabs/massa-as-sdk';
+import { Args } from '@massalabs/as-types';
+
+/**
+ * This function is meant to be called only one time: when the contract is deployed.
+ *
+ * @param _ - not used
+ */
+export function constructor(_: StaticArray<u8>): StaticArray<u8> {
+  // This line is important. It ensures that this function can't be called in the future.
+  // If you remove this check, someone could call your constructor function and reset your smart contract.
+  if (!callerHasWriteAccess()) {
+    return [];
+  }
+  const nBlogPosts = 0;
+  // Set the initial value of N_BLOG_POSTS to 0 in the storage of the contract
+  Storage.set("N_BLOG_POSTS", nBlogPosts.toString());
+  // This function is used to emit an event to the blockchain
+  generateEvent(`Blog initiated`);
+  return [];
+}
+
+export function post(binaryArgs: StaticArray<u8>): void {
+  const argsDeser = new Args(binaryArgs);
```

Comment:
pratice is to use _args and args:

```suggestion
export function post(_args: StaticArray<u8>): void {
  const args = new Args(_args);
```